docs/zh_cn/design/infer_translation.md

```diff
@@ -0,0 +1,244 @@
+
+To optimize the user experience of the inferencer, we aim to eliminate the need for users to configure parameters for each step when performing inference. In other words, we want users to be able to simply configure parameters for the `__call__` interface without having to be aware of the aforementioned processes.
+
+The `__call__` interface will execute the aforementioned steps in order, but it is not aware of which step the parameters provided by the user should be distributed to. Therefore, when developing a `CustomInferencer`, developers need to define four class attributes: `preprocess_kwargs`, `forward_kwargs`, `visualize_kwargs`, and `postprocess_kwargs`. Each attribute is a set of strings that are used to specify which step the parameters in the `__call__` interface correspond to:
+
+```python
+class CustomInferencer(BaseInferencer):
+    preprocess_kwargs = {'a'}
+    forward_kwargs = {'b'}
+    visualize_kwargs = {'c'}
+    postprocess_kwargs = {'d'}
+
+    def preprocess(self, inputs, batch_size=1, a=None):
+        pass
+
+    def forward(self, inputs, b=None):
+        pass
+
+    def visualize(self, inputs, preds, show, c=None):
+        pass
+
+    def postprocess(self, preds, visualization, return_datasample=False, d=None):
+        pass
+
+    def __call__(
+        self,
+        inputs,
+        batch_size=1,
+        show=True,
+        return_datasample=False,
+        a=None,
+        b=None,
+        c=None,
+        d=None):
+        return super().__call__(
+            inputs, batch_size, show, return_datasample, a=a, b=b, c=c, d=d)
+```
+
+In the code above, `a`, `b`, `c`, and `d` in the `preprocess`, `forward`, `visualize`, and `postprocess` functions are additional parameters that can be passed in by the user (`inputs`, `preds`, and other parameters are automatically filled in during the execution of `__call__`). Therefore, developers need to specify these parameters in the `preprocess_kwargs`, `forward_kwargs`, `visualize_kwargs`, and `postprocess_kwargs` class attributes, so that the parameters passed in by the user in the `__call__` phase can be correctly distributed to the corresponding steps. The distribution process is implemented by the `BaseInferencer.__call__` function, which developers do not need to be concerned about.
+
+In addition, we need to register the `CustomInferencer` to a custom registry or the MMEngine's registry.
+
+```python
+from mmseg.registry import INFERENCERS
+# It can also be registered to the registry of MMEngine.
+# from mmengine.registry import INFERENCERS
+
+@INFERENCERS.register_module()
+class CustomInferencer(BaseInferencer):
+    ...
+```
+
+```{note}
+In OpenMMLab's algorithm repositories, the Inferencer must be registered to the downstream repository's registry instead of the root registry of MMEngine to avoid naming conflicts.
+```
+
+**Core Interface Explanation**:
```

Comment:
It could be better to make it as a level 2 heading. Please also refine the Chinese doc

Reply:
Excuse me, is the English translation also named 'infer.md'? Will it conflict with Chinese doc?